app/graphql/schema/scalars/string-or-array.js

```diff
@@ -0,0 +1,45 @@
+import { GraphQLScalarType, Kind } from 'graphql'
+
+const invalidTypeError = 'Value must be either a string or an array of strings'
+const validate = (value) => {
+  if (typeof value === 'string' || (Array.isArray(value) && value.every(item => typeof item === 'string'))) {
+    return { value }
+  }
+}
+
+const StringOrArray = new GraphQLScalarType({
+  name: 'StringOrArray',
+  description: 'Custom scalar that can be either a string or an array of strings',
+  parseValue (value) {
+    const validatedValue = validate(value)
+    if (validatedValue === undefined) {
+      throw new Error(invalidTypeError)
+    }
+    return validatedValue.value
+  },
+  serialize (value) {
+    const validatedValue = validate(value)
+    if (validatedValue === undefined) {
+      throw new Error(invalidTypeError)
```

Comment:
We are always checking the `validatedValue` and throwing if it's not right. Could we not move this into the validate function?